wave_lang/kernel/wave/water.py

```diff
@@ -173,26 +174,72 @@ def replace_ops_and_collect_subspans(op: Operation) -> WalkResult:
     return local_module.get_asm(binary=False, print_generic_op_form=True)
 
 
-def find_binary(name: str) -> str:
+def find_binary(name: str) -> Optional[str]:
```

Comment:
```suggestion
def find_binary(name: str) -> str | None:
```

`Optional` is deprecated

Reply:
indeed, I'm just merging that into the same bucket as `List` and `Dict`